helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -1114,4 +1136,12 @@ public void unsubscribeChildChanges(String path, IZkChildListener childListener)
   public void reset() {
     // Nothing to do
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void close() {
+    _zkClient.close();
```

Comment:
Could you please add a null check and isClosed() check (if available)?

Reply:
done.